CONTRIBUTING.md

```diff
@@ -31,8 +31,15 @@ examples.
 
 Pull request
 ------------
-Provide the same information as you would for a [feature request](#feature-request). Also briefly
-describe what the implementation might look like.
+Provide the same information as you would for a [feature request](#feature-request).
+
+We ask that you:
+
+* [Write a good commit message](http://chris.beams.io/posts/git-commit/)
+* Issue 1 Pull Request per feature. Don't lump unrelated changes together.
+* [Squash all commits down into a single logical commit](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request) for your PR.
```

Comment:
Not sure I agree about squashing.  There are sort of two facets at play here:

1. Some features may be non-trivial and require several logical "steps", and I'd argue that each step should be a commit.  However, I may be in the minority on this.
2. When a PR is reviewed and commented on, I think it's best not to squash history that correlates with the review history.  This way, looking at the git history makes sense alongside the PR discussion.  It also makes sure that people in the middle of a review don't have to "start over" from the beginning - when you squash your fixes into the single commit, it conflates the history so that you have to look at the entire PR at once.